packages/pluggableWidgets/color-picker-web/src/components/ColorPicker.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { createElement, ReactElement, useEffect, useState, useCallback } from "react";
+import { Alert } from "@mendix/piw-utils-internal/components/web";
+import { ModeEnum, TypeEnum, DefaultColorsType, FormatEnum } from "../../typings/ColorPickerProps";
+import {
+    ColorState,
+    SketchPickerProps,
+    ChromePickerProps,
+    TwitterPickerProps,
+    HuePickerProps,
+    BlockPickerProps,
+    GithubPickerProps,
+    CirclePickerProps,
+    SwatchesPickerProps,
+    SliderPickerProps,
+    MaterialPickerProps,
+    CompactPickerProps
+} from "react-color";
+import classNames from "classnames";
+import { getColorPicker, parseColor, validateColorFormat } from "../utils";
+import { Input } from "./Input";
+import { Button } from "./Button";
+
+export interface ColorPickerProps {
+    id: string;
+    name: string;
+    tabIndex?: number | undefined;
+    onChange: () => void;
+    onChangeComplete: (value: string) => void;
+    mode: ModeEnum;
+    type: TypeEnum;
+    color: string;
+    defaultColors: DefaultColorsType[];
+    format: FormatEnum;
+    disabled: boolean;
+    invalidFormatMessage?: string;
+}
+
+export type ColorPickerConfigProps =
+    | SketchPickerProps
+    | ChromePickerProps
+    | TwitterPickerProps
+    | HuePickerProps
+    | BlockPickerProps
+    | GithubPickerProps
+    | CirclePickerProps
+    | SwatchesPickerProps
+    | SliderPickerProps
+    | MaterialPickerProps
+    | CompactPickerProps;
+export const ColorPicker = (props: ColorPickerProps): ReactElement => {
+    const defaultColor = {
+        hex: "#FFFFFF",
+        rgb: "rgb(255,255,255)",
+        rgba: "rgb(255,255,255,1)"
+    };
+    const { type, mode, disabled, defaultColors, color, format, invalidFormatMessage } = props;
+    const colorElement = getColorPicker(type);
+    const [hidden, setHidden] = useState(mode !== "inline");
+    const [config, setConfig] = useState<ColorPickerConfigProps>({});
+    const [currentColor, setCurrentColor] = useState<string>(color || defaultColor[format]);
+    const [alertMessage, setAlertMessage] = useState<string | undefined>();
+
+    const validateColor = (color: string): void => {
+        const validFormat = validateColorFormat(color, format);
+        const alertMessage = validFormat ? invalidFormatMessage?.replace(/\{1}/, validFormat) : undefined;
+        setAlertMessage(alertMessage);
+    };
+    const onToggle = useCallback(
+        (hide: boolean): void => {
+            if (!disabled && mode !== "inline") {
+                setHidden(hide);
```

Comment:
Should this be `!hide` since it's a toggle?

Reply:
I guess naming is little bit confusing, but there is a case when I need to just make sure that popup is closed, so I combined onClose and onToggle methods into a one, but didn't change the name, anyway, from my point of view when you see argument passed into a function it means that it should do something, so in this case it's toggling with a parameter which is passed as argument

Reply:
We normally use "toggle" when a boolean has to be inverted, which is not the case here. It's perfectly fine to combine `onClose` and `onToggle`, but the function name is confusing now as you mentioned. Something like `setColorPickerHidden` (since you can't use setHidden) or `changeHidden` would make more sense

Reply:
maybe `setColorPickerVisibility`?